node/src/test/scala/com/wavesplatform/api/eth/EthRpcRouteSpec.scala

```diff
@@ -41,6 +41,14 @@ class EthRpcRouteSpec extends RouteSpec("/eth") with WithDomain with EthHelpers
     routeTest(d, "eth_blockNumber")(resultInt shouldBe 11)
   }
 
+  "eth_getBlockByNumber" in withDomain() { d =>
+    1 to 5 foreach (_ => d.appendBlock())
+    routeTest(d, "eth_getBlockByNumber", "0x123")((resultJson \ "number").as[String] shouldBe "0x123")
+    routeTest(d, "eth_getBlockByNumber", "earliest")((resultJson \ "number").as[String] shouldBe "0x1")
+    routeTest(d, "eth_getBlockByNumber", "latest")((resultJson \ "number").as[String] shouldBe "0x5")
+    routeTest(d, "eth_getBlockByNumber", "pending")((resultJson \ "number").as[JsValue] shouldBe JsNull)
```

Comment:
If `pending` logic changes to be the same as `latest`
```suggestion
    routeTest(d, "eth_getBlockByNumber", "pending")((resultJson \ "number").as[JsValue] shouldBe "0x5")
```